eskip/lexer.go

```diff
@@ -125,13 +126,47 @@ func scanVoid(code string, p charPredicate) string {
 }
 
 func scanEscaped(delimiter byte, code string) ([]byte, string) {
-	var b []byte
+	var (
+		b              []byte
+		bin            [2]byte
+		secondBinValue bool
+	)
 	escaped := false
+	binary := false
 	for len(code) > 0 {
 		c := code[0]
 		isDelimiter := c == delimiter
 		isEscapeChar := c == escapeChar
 
+		if binary {
+			switch c {
+			case delimiter:
+				binary = false
+			case escapeChar:
+				binary = false
+			case 'x':
+				code = code[1:]
+				continue
+			default:
+				if !secondBinValue {
+					secondBinValue = true
+					i, err := strconv.ParseInt(string(c), 16, 8)
+					if err != nil {
+						break
+					}
+					bin[0] = byte(i)
+				} else {
+					secondBinValue = false
+					i, err := strconv.ParseInt(string(c), 16, 8)
+					if err != nil {
+						binary = false
+						code = code[1:]
+						continue
+					}
+					bin[1] = byte(i)
+				}
+			}
```

Comment:
I am not sure it is actually working. I do not see we store parsed binary bytes into returned result `b`  anywehere which is confirmed by this test that fails:
```go
func TestStringHex(t *testing.T) {
	for _, tc := range []struct {
		filter, expected string
	}{
		{`foo("\x00")`, "\x00"},
		{`foo("a\x00")`, "a\x00"},
		{`foo("\x00b")`, "\x00b"},
		{`foo("a\x00b")`, "a\x00b"},

		{`foo("\x000")`, "\x000"},
	} {
		t.Run(tc.filter, func(t *testing.T) {
			ff, err := ParseFilters(tc.filter)
			if err != nil {
				t.Fatal(err)
			}

			s := ff[0].Args[0]

			if s != tc.expected {
				t.Errorf("wrong string value, expected %q: got: %q", tc.expected, s)
			}
		})
	}
}
```

We also need to adjust eskip serialization logic to escape binary values somehow I guess.

Reply:
how do you think the filter blockContent would be able to block the tested content?